jerry-core/ecma/operations/ecma-bigint.c

```diff
@@ -0,0 +1,508 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ecma-bigint.h"
+#include "ecma-big-uint.h"
+#include "ecma-exceptions.h"
+#include "ecma-helpers.h"
+#include "lit-char-helpers.h"
+
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+
+/**
+ * Raise a not enough memory error
+ *
+ * @return ECMA_VALUE_ERROR
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_bigint_raise_memory_error (void)
+{
+  return ecma_raise_range_error (ECMA_ERR_MSG ("Cannot allocate memory for a BigInt value"));
```

Comment:
Generally we used to terminate the engine when we run out of memory, why do we handle bigints differently?

Reply:
I think this is useful feature, and it does not need too much effort, similar to parser.